vcpkg/concepts/registries.md

```diff
@@ -3,49 +3,404 @@ title: Registries concepts
+The built-in registry is a special kind of registry. The built-in registry
+is the default registry used in some operation modes, like [classic
+mode](classic-mode.md). When a [default
+registry](../reference/vcpkg-configuration-json.md#default-registry) is not
+specified, vcpkg implicity uses the built-in registry as the default.
 
-If vcpkg was acquired using `git clone`, this will refer to the registry in `VCPKG_ROOT` itself,
-and is expected to be a clone of <https://github.com/Microsoft/vcpkg> created before running vcpkg.
+The built-in registry most often refers to the local copy of the curated
+registry created when you `git clone` the vcpkg repository from
+<https://github.com/Microsoft/vcpkg>. Some operations expect that the
+[`VCPKG_ROOT` environment variable](../users/config-environment.md#vcpkg_root)
+points to a built-in registry.
 
-Otherwise (vcpkg was acquired using the 'one liner' installer or the 'Visual Studio bundle'),
-this will be equivalent to a git registry with a `"repository"` of
-`"https://github.com/Microsoft/vcpkg"`.
+If vcpkg is acquired through the "one liner" or Visual Studio installer, the
+built-in registry becomes equivalent to a [Git registry](#git-registries)
+pointing to the <https://github.com/Microsoft/vcpkg> repository.
 
 ## Git registries
 
-Git registries are simple Git repositories. They can be shared publicly or
-privately via normal mechanisms for Git repositories. The [vcpkg
-repository](https://github.com/microsoft/vcpkg) is an example of a Git registry.
+A Git registry is a repository that follows the registry structure and leverages
+Git's capatilities to provide versioning for the ports in the registry. The
+curated registry at <https://github.com/Microsoft/vcpkg> is an implementation of
+a Git registry.
+
+Git registries can be hosted in any Git repository provider, allowing you to
+use your chosen Git hosting service to control access to your custom registry
+while also making your registry easy to share.
 
-Using Git registries offers the best experience for custom registries since you
-have full control over the versions and contents of your registry.
+Git registries are the recommended method for implementing a custom registry.
+Since the versioning mechanism is the same used by the curated registry, Git
+registries can make use of the [`x-add-version`](../commands/add-version.md) to
+manage your version database files.
+
+See the [registries reference](../maintainers/registries.md#git-registries) for
+implementation details of Git registries.
+
+### <a name="git-add-version"> Adding a new version to a Git registry
+
+The [`x-add-version`](../commands/add-version.md) command can be used to to add
+a new port or a new version to the registry. When adding versions using this
+command there are a couple of things to keep in mind:
+
+**Remember to update the port's version**
+
+> [!IMPORTANT]
+> Always remember to update the port's version to avoid rewriting the version
+> history.
```

Comment:
It seems like the bolded bit and the important block say the same thing?

```suggestion
> [!IMPORTANT]
> When adding a new version, always remember to update the port's declared version to one not already published, to avoid rewriting the version history.
```

I'm not so sure my attempt in the suggestion is an improvement; the point I'm trying to describe is that `--overwrite-version` may be appropriate when the version being edited isn't yet published.